tests/mocks/files.py

```diff
@@ -0,0 +1,56 @@
+import json
+import os
+from pathlib import Path
+
+
+class MockEntityFiles:
+    def __init__(self, base_path=None, base_uri=None):
+        self.base_path = base_path
+        if self.base_path is None:
+            self.base_path = os.path.dirname(__file__) + '/ingest-data/'
+        self.base_uri = base_uri
+        if self.base_uri is None:
+            self.base_uri = 'http://mock-ingest-api/'
+
+    def get_entity(self, entity_type, entity_id):
+        self_location = Path(self.base_path + entity_type + '/' + entity_id + '/self.json')
+        if self_location.exists() and self_location.is_file():
+            return self.load_json_file(self_location)
+        location = Path(self.base_path + entity_type + '/' + entity_id + '.json')
+        if location.exists() and location.is_file():
+            return self.load_json_file(location)
+
+    def get_related_entities(self, base_entity_uri, relation_endpoint) -> list:
+        base_type_id = base_entity_uri.replace(self.base_uri, '')
+        base_type = base_type_id.split('/')[0]
+        base_id = base_type_id.split('/')[1]
+        relation_name = str.strip(relation_endpoint.replace(base_entity_uri, ''), '/')
+
+        full_path = Path(self.base_path + base_type + '/' + base_id + '/' + relation_name)
+        if full_path.exists() and full_path.is_file():
+            return self.get_entities_from_file(full_path)
+        if full_path.exists() and full_path.is_dir():
+            return self.load_json_files_from_dir(full_path)
+        return []
+
+    def get_entities_from_file(self, location: Path) -> list:
```

Comment:
I think `Iterator` (from `from typing import Iterator`) is the correct return type here. If this returned a list I'd expect `get_entities_from_file[0]` to work. Looks like pycharm doesn't catch this but `mypy` does